nestkernel/event_delivery_manager.cpp

```diff
@@ -571,6 +571,27 @@ EventDeliveryManager::deliver_events_( const thread tid, const std::vector< Spik
 
         kernel().connection_manager.send( tid, syn_id, lcid, cm, se );
       }
+      else if ( spike_data.get_tid() == MAX_TID )  // handle compressed spikes
+      {
+        se.set_stamp( prepared_timestamps[ spike_data.get_lag() ] );
+        se.set_offset( spike_data.get_offset() );
+
+        const index syn_id = spike_data.get_syn_id();
+        const index idx = spike_data.get_lcid();  // for compressed spikes lcid holds the index in the compressed_spike_data structure
+        const std::vector< SpikeData >& compressed_spike_data =
+          kernel().connection_manager.get_compressed_spike_data( syn_id, idx );
+        for ( auto it = compressed_spike_data.cbegin(); it != compressed_spike_data.cend(); ++it )
```

Comment:
Is type inference used by NEST? I can't see any other instance of `auto` elsewhere. Explicitly state type for consistency and clarity.

Reply:
@7eia NEST is older than the C++-98 standard, our first commit dates back to 1996. We are phasing in C++11 constructs in new or refactored code, but do not have the capacity to refactor everything to C++11.

Reply:
@heplesser Noted.